src/main/kotlin/com/jaoafa/vcspeaker/tts/Scheduler.kt

```diff
@@ -69,14 +100,28 @@ class Scheduler(
 
     override fun onTrackEnd(player: AudioPlayer, track: AudioTrack, endReason: AudioTrackEndReason): Unit =
         runBlocking {
+            val info = track.userData as SpeakInfo
+            val message = info.message
+            val guildName = info.guild.name
+
             launch {
-                (track.userData as SpeakInfo).message?.deleteOwnReaction(ReactionEmoji.Unicode("🔊"))
+                message?.deleteOwnReaction(ReactionEmoji.Unicode("🔊"))
             }
 
             if (endReason.mayStartNext && queue.isNotEmpty()) {
                 now = queue.removeFirst()
                 launch { speak(now!!) }
-            } else now = null
+
+                logger.info {
+                    "[$guildName] Next Track Starting: Audio track for ${info.getMessageLogInfo()} popped out from the queue."
```

Comment:
これ和訳なんですか？ "<メッセージ>のトラックがキューから飛び出しました" …？
array.pop的なニュアンスだと思うんですが、"Starts playing the next track <メッセージ> from the queue." とかでもいいのではーと思ったり

Reply:
ここは Queue から取り出した段階で、Play されることが確約されているわけではないのでそういう風に書きました。

Reply:
あー…、onTrackEndだから次のトラックが確実に再生されたわけではないですね（されうるとしても）
ただそうである場合、この段階ではpopしているわけではないということにもなるので、「再生が完了した」「次のトラックがある」というニュアンスのログが正しいかなと思います。（`Next Track Starting`も怪しいですよね）

以下提案です。

```suggestion
                    "[$guildName] Playing Track Finished: There is a next track in the queue. It can be played when it is ready to be played."
```

Reply:
pop はしてます (`now = queue.removeFirst()`)
"Next Track Starting" は "次のトラックが始まろうとしている" という意味で怪しくはないはずです。
ただ popped out はやっぱり has been retrieved とかのほうがいいかなと思い始めました。

Reply:
そうか…🤔popped out の書き方については、まかせます